Bio/Seq.py

```diff
@@ -1953,6 +2049,57 @@ def __radd__(self, other):
         else:
             raise TypeError
 
+    def __mul__(self, other):
+        """Multiply MutableSeq by integer
+
+        Note this is not in-place and returns a new object,
+        matching native Python list multiplication
+
+        >>> from Bio.Seq import MutableSeq
+        >>> from Bio.Alphabet import generic_dna
+        >>> MutableSeq('ATG') * 2
+        MutableSeq('ATGATG', Alphabet())
+        >>> MutableSeq('ATG', generic_dna) * 2
+        MutableSeq('ATGATG', DNAAlphabet())
+        """
+
+        if not isinstance(other, int):
+            raise TypeError("can't multiply MutableSeq by non-int type")
+        return self.__class__(str(self) * other, self.alphabet)
```

Comment:
I wonder if we can make this much more efficient by working directly with the ``self.data`` array, avoiding conversion to and from a string?